### PR TITLE
Feature/products

### PR DESCRIPTION
**Pull request by kool7**

### CODE REVIEW

File: tests/Unit/ProductHub.UnitTests/Controller/ProductTestController.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        var products = _fixture.CreateMany<Product>();
+        _mockProductsService.Setup(service => service.GetProductsAsync())
+            .ReturnsAsync(products);
+
+        // Act
+        var result = await _sutProductsController.GetProductsAsync();
+
+        // Assert
+        result.Result.Should().BeOfType<OkObjectResult>();
+        result.Value?.Count().Should().Be(products.Count());
```
Comment (rohitvipin) [inline tests/Unit/ProductHub.UnitTests/Controller/ProductTestController.cs]: testing for count is not good enough, test for actual v expected.

File: tests/Unit/ProductHub.UnitTests/Controller/ProductTestController.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        _mockProductsService = new Mock<IProductsService>();
+
+        if (_mockMapper == null)
+        {
+            var mappingConfig = new MapperConfiguration(mc =>
+            {
+                mc.AddProfile(new ProductHubAutoMapperProfile());
+            });
+            IMapper mapper = mappingConfig.CreateMapper();
+            _mockMapper = mapper;
```
Comment (rohitvipin) [inline tests/Unit/ProductHub.UnitTests/Controller/ProductTestController.cs]: this is not mock, this is actual mapper

File: tests/Unit/ProductHub.UnitTests/Controller/ProductTestController.cs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        var product = _fixture.Create<Product>();
+        _mockProductsService
+            .Setup(service => service.GetProductAsync(product.Id!))
+            .ReturnsAsync(product);
+
+        // Act
+        var result = await _sutProductsController.GetProductAsync(product.Id!);
+
+        // Assert
+        result.Should().BeOfType<OkObjectResult>();
```
Comment (rohitvipin) [inline tests/Unit/ProductHub.UnitTests/Controller/ProductTestController.cs]: get should be tested against actual

**Review by rohitvipin — CHANGES_REQUESTED** [review]

File: tests/Unit/ProductHub.UnitTests/Controller/ProductTestController.cs (hunk 3 above)
Comment (kool7) [reply tests/Unit/ProductHub.UnitTests/Controller/ProductTestController.cs]: Okay. Working on fix right now.

File: src/ProductHub.Api/Data/ProductDbContext.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        private readonly IMongoDatabase _database;
+
+        public ProductDbContext(IOptions<MongoDBSettings> options)
+        {
+            var connectionString = options.Value.ConnectionURI;
+            var client = new MongoClient(connectionString);
+            _database = client.GetDatabase(options.Value.DatabaseName);
+        }
+
+        public IMongoCollection<Product> Products => _database.GetCollection<Product>("products");
```
Comment (aashikvilla) [inline src/ProductHub.Api/Data/ProductDbContext.cs]: Collection name is part of MongoDbSettings , you can use that

**Pull request merged**